go_covert_lib/controller/processor/symmetricEncryption/symmetricEncryption.go

```diff
@@ -0,0 +1,237 @@
+		block     cipher.Block
+		err       error
+	)
+
+	// based on the users choice of symmetric algorithm create a cipher
+	switch algorithm {
+	case "Advanced Encryption Standard (AES)":
+		block, err = aes.NewCipher(c.key)
+		blockSize = 16
+	case "Data Encryption Standard (DES)":
+		block, err = des.NewCipher(c.key)
+		blockSize = 8
+	case "Triple Data Encryption Standard (3DES)":
+		block, err = des.NewTripleDESCipher(c.key)
+		blockSize = 8
+	default:
+		return nil, errors.New("Undefined algorithm selected")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	// based on the users choice of the mode of operation encrypt in that mode
+	var cipherText []byte
+	selectedMode := c.mode
+	switch selectedMode {
+	case "Galois Counter Mode (GCM)":
+		cipherText = GCMEncrypter(block, data)
+		if cipherText == nil {
+			return nil, errors.New("Unable to encrypt in Galosis 
+									Counter Mode (GCM)")
+		}
+	case "Cipher Block Chaining (CBC)":
+		cipherText = CBCEncrypter(block, data, blockSize)
+	case "Cipher Feedback (CFB)":
+		cipherText = CFBEncrypter(block, data, blockSize)
+	case "Counter (CTR)":
+		cipherText = CTREncrypter(block, data, blockSize)
+	default:
+		return nil, errors.New("Undefined mode selected")
+	}
+
+	return cipherText[:], nil
+}
+
+func GCMEncrypter(block cipher.Block, data []byte) []byte {
+	nonce := make([]byte, 12)
+
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil
+	}
+
+	cipherText := aesgcm.Seal(nil, nonce, data, nil)
+
+	return cipherText[:]
+}
+
+func CBCEncrypter(block cipher.Block, data []byte, blockSize int) []byte {
+	cipherText := make([]byte, blockSize+len(data))
+	iv := cipherText[:blockSize]
+
+	mode := cipher.NewCBCEncrypter(block, iv)
+	mode.CryptBlocks(cipherText[blockSize:], data)
+
+	return cipherText[:]
+}
+
+func CFBEncrypter(block cipher.Block, data []byte, blockSize int) []byte {
+	cipherText := make([]byte, blockSize+len(data))
+	iv := cipherText[:blockSize]
+
+	stream := cipher.NewCFBEncrypter(block, iv)
+	stream.XORKeyStream(cipherText[blockSize:], data)
+
+	return cipherText[:]
+}
+
+func CTREncrypter(block cipher.Block, data []byte, blockSize int) []byte {
+	cipherText := make([]byte, blockSize+len(data))
+	iv := cipherText[:blockSize]
+
+	stream := cipher.NewCTR(block, iv)
+	stream.XORKeyStream(cipherText[blockSize:], data)
+
+	return cipherText[:]
+}
+
+func Pad(data []byte) []byte {
+	// padding is done by having 7 zero's at the start followed by the length 
+	// of the original message, the contents of the message, followed by a 
+	// series of zero's to fill the block size requirements of the symmetric  
+	// encryption algorithm 
+	lenOfData := len(data)
+	data = append(data, 0)
+	copy(data[1:], data)
+	data[0] = byte(lenOfData)
+
+	for i := 0; i < padIndex; i++ {
+		data = append(data, 0)
+		copy(data[1:], data)
+		data[0] = 0
+	}
+
+	for len(data)%aes.BlockSize != 0 {
+		data = append(data, 0)
+	}
+
+	return data[:]
+}
+
+func UnPad(data []byte) []byte {
```

Comment:
Again, you will have to modify this to recognise that more than 255 bytes could be sent.